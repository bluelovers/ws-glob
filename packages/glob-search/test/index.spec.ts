import { globSearchSync, globSearch } from '../index';

test(`globSearch`, async () =>
{

	let actual = await globSearch('*/demo.ts', {
			disableThrowWhenEmpty: true,
		})
		.tap(function (data)
		{
			console.log(data);
			expect(data.value[0]).toStrictEqual('test/demo.ts')
		})
		.catch(console.error)
	;

	expect(actual).toMatchSnapshot({
		cwd: expect.any(String),
		history: expect.any(Array),
		options: expect.any(Object),
	});

});

test(`globSearchSync`, () =>
{

	let actual = globSearchSync('*/demo.ts', {
			disableThrowWhenEmpty: true,
		})
		.catch(console.error)
		.then(v => v)
		.tap(function (data)
		{
			console.log(data);
			expect(data.value[0]).toStrictEqual('test/demo.ts')
		})
	;

	expect(actual).toMatchSnapshot({
		cwd: expect.any(String),
		history: expect.any(Array),
		options: expect.any(Object),
	});

});
