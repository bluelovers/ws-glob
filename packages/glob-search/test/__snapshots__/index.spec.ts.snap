// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globSearch 1`] = `
{
  "cwd": Any<String>,
  "history": Any<Array>,
  "options": Any<Object>,
  "pattern": [
    "*/demo.ts",
  ],
  "value": [
    "test/demo.ts",
  ],
}
`;

exports[`globSearchSync 1`] = `
{
  "cwd": Any<String>,
  "history": Any<Array>,
  "options": Any<Object>,
  "pattern": [
    "*/demo.ts",
  ],
  "tapCatch": [Function],
  "value": [
    "test/demo.ts",
  ],
}
`;
